sdk/communication/Azure.Communication.Chat/tests/ChatClients/ChatClientsTest.cs

```diff
@@ -84,22 +91,74 @@ public void E2E_ThreadCreateUpdateGetDelete_MemberAddUpdateRemove_MessageGetSend
+            var messageContent2 = "Content for message 2";
+            SendChatMessageResult sendChatMessageResult2 = chatThreadClient2.SendMessage(messageContent2, ChatMessagePriority.High, displayNameMessage);
+            var messageContent3 = "Content for message 3";
+            SendChatMessageResult sendChatMessageResult3 = chatThreadClient3.SendMessage(messageContent3, ChatMessagePriority.High, displayNameMessage);
+            var messageContent4 = "Content for message 4";
+            SendChatMessageResult sendChatMessageResult4 = chatThreadClient3.SendMessage(messageContent4, ChatMessagePriority.High, displayNameMessage);
+            var messageContent5 = "Content for message 5";
+            SendChatMessageResult sendChatMessageResult5 = chatThreadClient3.SendMessage(messageContent5, ChatMessagePriority.High, displayNameMessage);
+            var messageContent6 = "Content for message 6";
+            SendChatMessageResult sendChatMessageResult6 = chatThreadClient3.SendMessage(messageContent6, ChatMessagePriority.High, displayNameMessage);
 
             var messageId = sendChatMessageResult.Id;
+            var messageId2 = sendChatMessageResult2.Id;
+            var messageId3 = sendChatMessageResult3.Id;
+            var messageId4 = sendChatMessageResult4.Id;
+            var messageId5 = sendChatMessageResult5.Id;
+            var messageId6 = sendChatMessageResult6.Id;
+
             #region Snippet:Azure_Communication_Chat_Tests_E2E_GetMessage
             ChatMessage message = chatThreadClient.GetMessage(messageId);
             #endregion Snippet:Azure_Communication_Chat_Tests_E2E_GetMessage
+            ChatMessage message2 = chatThreadClient2.GetMessage(messageId2);
+            ChatMessage message3 = chatThreadClient3.GetMessage(messageId3);
+            ChatMessage message4 = chatThreadClient3.GetMessage(messageId4);
+            ChatMessage message5 = chatThreadClient3.GetMessage(messageId5);
+            ChatMessage message6 = chatThreadClient3.GetMessage(messageId6);
 
             #region Snippet:Azure_Communication_Chat_Tests_E2E_GetMessages
             Pageable<ChatMessage> messages = chatThreadClient.GetMessages();
             #endregion Snippet:Azure_Communication_Chat_Tests_E2E_GetMessages
+            Pageable<ChatMessage> messages2 = chatThreadClient2.GetMessages();
             var getMessagesCount = messages.Count();
+            var getMessagesCount2 = messages2.Count();
+
+            # region Pagination assertions
+            Pageable<ChatMessage> messagesPaginationTest = chatThreadClient.GetMessages();
+            string? continuationToken = null;
+            var expectedPageSize = 2;
+            var messagesCounterTotal = 0;
+            var messagesCounter = 0;
+            foreach (Page<ChatMessage> messagesPage in messagesPaginationTest.AsPages(continuationToken,2))
+            {
+                messagesCounter = 0;
+                foreach (ChatMessage messagePage in messagesPage.Values)
```

Comment:
Can we rename and simplify these two counters?

Use `var actualPageSize = messagePage.Values.Count()` instead of the loop on line 140
Keep a `messagesCount` outside the loop of line 137 and  `+=`  the count, on line 153 assert on `actualPageSize`

Reply:
I actually wanted to go through each of the items in each of the pages to exercise the iterators in both. Any concerns ?

Reply:
At this point you're testing the enumerable implemtation of  `ReadonlyList`? Not sure if you're gaining anything from doing that.